scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -91,7 +92,44 @@ impl ReplicatorConsumer {
         Ok(())
     }
 
-    // Recreates INSERT/DELETE statement on non-frozen map.
+    // Function replicates adding and deleting elements
+    // using v = v + {} and v = v - {} cql syntax to non-frozen map.
+    // Chained operations are supported.
+    async fn update_map_elements<'a>(
+        &self,
+        column_name: &str,
+        data: &'a CDCRow<'_>,
+        values_for_update: &mut Vec<&'a CqlValue>,
+        timestamp: i64,
+    ) -> anyhow::Result<()> {
+        let empty_map = Map(vec![]); // We have to declare it, because we can't take a reference to tmp value in unwrap.
+        let value = data.get_value(column_name).as_ref().unwrap_or(&empty_map);
+        // Order of values: ttl, added elements, pk condition values.
+
+        let deleted_set = Set(Vec::from(data.get_deleted_elements(column_name)));
```

Comment:
Another opportunity for later: here we are creating a vector from a slice, which is also a form of copying. We could take a mutable reference to the CDCRow and take deleted elements out of it.